aten/src/ATen/native/ForeachUtils.h

```diff
@@ -137,8 +143,18 @@ bool can_use_fast_route(TensorList tensors, Scalar scalar) {
 #endif
 }
 
-bool can_use_fast_route(TensorList tensors, ArrayRef<double> scalars) {
-  return can_use_fast_route(tensors);
+bool can_use_fast_route(TensorList tensors, ArrayRef<Scalar> scalars) {
+#ifdef __HIP_PLATFORM_HCC__
+  return false;
+#else
+  for (int i = 0; i < tensors.size(); i++) {
+    if (will_promote_tensor(tensors[i], scalars[i])) {
+      return false;
+    }
+  }
+
+  return true;
+#endif
```

Comment:
Do we need to use has_same_attributes on `tensors`? E.g. to check that all tensors have the same device and are all strided?

Reply:
For all the APIs, im calling `has_same_attributes` first and then `can_use_fast_route`. The reason for that is that these two APIs are checking for different thing so i didn't want to introduce confusion.